src/event.js

```diff
@@ -366,40 +366,41 @@ jQuery.event = {
 		//
 		// Support: Firefox <=42
 		// Avoid non-left-click in FF but don't block IE radio events (#3861, gh-2343)
-		if ( delegateCount && cur.nodeType &&
-			( event.type !== "click" || isNaN( event.button ) || event.button < 1 ) ) {
+		if ( delegateCount && cur.nodeType && !( event.type === "click" && event.button >= 1 ) ) {
```

Comment:
I know this is shorter but it isn't as obvious. However, I can't recall the browser where `event.button` doesn't exist nowadays though. It seems to be there on IE9+ but maybe Android or something? If that's not a common case I can't really object to this change since it may no longer exist.